apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -3094,7 +3055,7 @@ impl OpGraphPath {
         // account (it may very well be that whatever comes after `u` is not in `A`, for instance).
         let self_tail_weight = self.graph.node_weight(self.tail)?;
         let other_tail_weight = self.graph.node_weight(other.tail)?;
-        if self_tail_weight.source == other_tail_weight.source {
+        if self_tail_weight.source != other_tail_weight.source {
```

Comment:
This change does not affect the linked ticket. This differs from the JS code and was found while trying to locate the bug that I was looking for.

Reply:
@TylerBloom Good catch! It's indeed flipped from the JS code.